healthcheck/ticker.go

```diff
@@ -36,7 +36,7 @@ func (ticker *ticker) start(ctx context.Context) {
 			case <-ticker.closing:
 				return
 			case <-ticker.timeTicker.C:
-				go ticker.runCheck(ctx)
```

Comment:
I don't know the original reason why we had a go function here. I imagine that it was so that we don't block the loop go-routine ?
If we need to make sure we don't block it, and still wait for it to finish until the checkers have been executed, I would suggest to do the following:
- add a wait group to the ticker struct
- in the start function:
```
go func() {
  for {
    ...
      case <-ticker.timeTicker.C:
      ticker.wg.Add(1)
      go ticker.runCheck(ctx)
...}
    ticker.wg.Wait()
}
```
- runCheck:
```
runCheck(){
   defer ticker.wg.Close()
   ...
}
```

Reply:
I think David is right, if we don't have the ticker.runCheck(ctx) in a go routine then when context times out and calls ctx.Done, it won't receive it and will block when the application is trying to shutdown (ungracefully at this point)